panel/lxqtpanel.cpp

```diff
@@ -215,8 +216,11 @@ LXQtPanel::LXQtPanel(const QString &configGroup, LXQt::Settings *settings, QWidg
     connect(qApp, &QApplication::screenRemoved, this, [this] (QScreen* oldScreen) {
         disconnect(oldScreen, &QScreen::virtualGeometryChanged, this, &LXQtPanel::ensureVisible);
         disconnect(oldScreen, &QScreen::geometryChanged, this, &LXQtPanel::ensureVisible);
-        // wait until the screen is really removed because it may contain the panel
-        QTimer::singleShot(0, this, &LXQtPanel::ensureVisible);
+        if (QGuiApplication::platformName() != QStringLiteral("wayland"))
```

Comment:
Could this be default but a setting to change for both x11 and wayland?

Reply:
I guess what you mean is monitor-specific panels for X11 too. That may be possible in the same way, but
 * I really prefer not to touch X11 here (in order to avoid any chance of regression);
 * There are some codes for X11 to forcefully move the panel, and I think there should have been a good reason for including them;
 * To work on X11, I should log into an X11 session, which is so annoying ;)

Maybe later, in another PR.

Reply:
That could also be an option, but I didn't meant this.
I meant that also on wayland panels _could_ move to the other monitor if their monitor disappears, optionally. Situations like people which sometimes work with only the laptop monitor and other times using only an external one, now they would need to recreate a panel.

Reply:
> Situations like people which sometimes work with only the laptop monitor and other times using only an external one

With this patch, they'll have no problem :) That panel(s) of each monitor will be created and shown as soon as it's connected. `LXQtPanelApplication::handleWaylandScreenAdded` is about that.

Reply:
Oh, your comment showed me that I should remove a case of `return;`. Doing it… Thanks!

Reply:
Not really... maybe you have 2 panels and want just them to go the big screen if you turn off your laptop screen.

Reply:
Done.

> maybe you have 2 panels and want just them to go the big screen if you turn off your laptop screen.

The new logic is that of screen-specific panels, which is at odds with the above. And there's no practical benefit to the latter.